scripts/im_calc_checkpoint.py

```diff
@@ -16,77 +24,92 @@
+    Check for any csv files in the given folder, or stations folder inside.
+    Check for any imcalc.info files.
+    If at least one of each type are present return True, otherwise return False
+    :param output_dir: The path to the IM_calc directory
+    :param station_count: The number of expected stations
+    :param event_name: The name of the event being checked. If not none assumes event was observed, rather than simulated.
+    :param verbose: Print extra information or not
+    :return: True if all conditions for job completion are true, false otherwise
+    """
     if os.path.isdir(output_dir):  # if output dir exists
 
         sum_csv = glob.glob1(output_dir, CSV_PATTERN)
-        meta = glob.glob1(output_dir, META_PATTERN)
-        # if sum_csv and meta are not empty lists('.csv' and '_imcalc.info' files present)
-        # then we think im calc on the corresponding dir is completed and hence remove
-        return (sum_csv and meta) != []
-
+        if len(sum_csv) > 0 and station_count > 0:
+            if event_name:
+                sum_csv = [f for f in sum_csv if event_name in f]
+                if not sum_csv:
+                    print_verbose("No csv file found", verbose)
+                    return False
 
-def checkpoint_wrapper(run_dir, waveform_dirs, sim_or_obs):
-    done = 0
-    todo = 0
-    for directory in waveform_dirs[:]:
-        dir_name = directory.split("/")[-1]
-        if dir_name == IM_CLAC_DIR:
-            waveform_dirs.remove(directory)
-        else:
-            exits = checkpoint_single(run_dir, dir_name, sim_or_obs)
-            if exits:
-                waveform_dirs.remove(directory)
-                done += 1
-            else:
-                todo += 1
-    if waveform_dirs[:] != []:
-        print("Inside {}, {} im_calc done, {} to do.".format(run_dir, done, todo))
-    return waveform_dirs
+            csv_file_name = sum_csv[0]
 
+            with open(os.path.join(output_dir, csv_file_name)) as csv_file:
+                if station_count != (len(csv_file.readlines()) - 1):
+                    print_verbose("CSV file does not have enough lines in it", verbose)
+                    return False
 
-def checkpoint_rrup(output_dir, srf_files):
-    """
-    Checkpoint for rrups
-    :param output_dir: user input output dir to store computed rrups
-    :param srf_files: a list of both completed and not completed srf files for computing rrups.
-    :return: a list of not completed srf files
-    """
-    for srf in srf_files[:]:
-        srf_name = srf.split("/")[-1].split(".")[0]
-        output_path = os.path.join(output_dir, srf_name + CSV_SUFFIX)
-        if os.path.isfile(output_path):
-            srf_files.remove(srf)
-    return srf_files
+        station_dir = os.path.join(output_dir, "stations")
+        if os.path.isdir(station_dir) and not event_name:
+            station_files = glob.glob1(station_dir, CSV_PATTERN)
+            if 0 < station_count != len(station_files):
```

Comment:
This either needs logic expansion or a comment. It's not intuitive what it does. I prefer the former.